http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpRStudioAppDAO.scala

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import cats.effect.Async
+import cats.mtl.Ask
+import cats.syntax.all._
+import org.broadinstitute.dsde.workbench.leonardo.AppContext
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.http4s._
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.headers.Authorization
+
+/**
+ * HTTP client for RStudio app.
+ */
+class HttpRStudioAppDAO[F[_]](httpClient: Client[F])(implicit
```

Comment:
Just thinking about this -- for GCP we abstracted all app status checking behind the common [HttpAppDAO](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDAO.scala). It's only for Azure that we started making DAOs per service -- partly because of the "combined" nature of many of the Azure apps (e.g. a single app instance exposes many services we need to status check).

I guess I'm wondering is `HttpRStudioAppDAO` actually needed or could we leverage `HttpAppDAO` for status checking?

Reply:
… I guess we'll use this when/if we support rstudio-app on Azure so it's probably good to have. 👍

Reply:
Yeah, I am thinking about our future app enablement work, it might be nice to have this be explicitly for Rstudio app